src/Contract/Prelude.purs

```diff
@@ -0,0 +1,324 @@
+import Data.Either
+  ( Either(Left, Right)
+  , choose
+  , either
+  , fromLeft
+  , fromLeft'
+  , fromRight
+  , fromRight'
+  , hush
+  , isLeft
+  , isRight
+  , note
+  , note'
+  ) as Either
+import Data.Enum
+  ( class BoundedEnum
+  , class Enum
+  , Cardinality(Cardinality)
+  , cardinality
+  , defaultCardinality
+  , defaultFromEnum
+  , defaultPred
+  , defaultSucc
+  , defaultToEnum
+  , downFrom
+  , downFromIncluding
+  , enumFromThenTo
+  , enumFromTo
+  , fromEnum
+  , pred
+  , succ
+  , toEnum
+  , toEnumWithDefaults
+  , upFrom
+  , upFromIncluding
+  ) as Enum
+import Data.Foldable
+  ( class Foldable
+  , all
+  , and
+  , any
+  , elem
+  , find
+  , findMap
+  , fold
+  , foldM
+  , foldMap
+  , foldMapDefaultL
+  , foldMapDefaultR
+  , foldl
+  , foldlDefault
+  , foldr
+  , foldrDefault
+  , indexl
+  , indexr
+  , intercalate
+  , length
+  , lookup
+  , maximum
+  , maximumBy
+  , minimum
+  , minimumBy
+  , notElem
+  , null
+  , oneOf
+  , oneOfMap
+  , or
+  , product
+  , sum
+  , surround
+  , surroundMap
+  ) as Foldable
+import Data.Maybe
+  ( Maybe(Just, Nothing)
+  , fromJust
+  , fromMaybe
+  , fromMaybe'
+  , isJust
+  , isNothing
+  , maybe
+  , maybe'
+  , optional
+  ) as Maybe
+-- A lot of this module is already re-exported from Data.Foldable
+import Data.Traversable
+  ( class Traversable
+  , Accum
+  , for
+  , for_
+  , mapAccumL
+  , mapAccumR
+  , scanl
+  , scanr
+  , sequence
+  , sequenceDefault
+  , sequence_
+  , traverse
+  , traverseDefault
+  , traverse_
+  ) as Traversable
+import Data.Tuple (Tuple(Tuple), curry, fst, snd, swap, uncurry) as Tuple
+import Data.Tuple.Nested
+  ( type (/\)
+  , T10
+  , T11
+  , T2
+  , T3
+  , T4
+  , T5
+  , T6
+  , T7
+  , T8
+  , T9
+  , Tuple1
+  , Tuple10
+  , Tuple2
+  , Tuple3
+  , Tuple4
+  , Tuple5
+  , Tuple6
+  , Tuple7
+  , Tuple8
+  , Tuple9
+  , curry1
+  , curry10
+  , curry2
+  , curry3
+  , curry4
+  , curry5
+  , curry6
+  , curry7
+  , curry8
+  , curry9
+  , get1
+  , get10
+  , get2
+  , get3
+  , get4
+  , get5
+  , get6
+  , get7
+  , get8
+  , get9
+  , over1
+  , over10
+  , over2
+  , over3
+  , over4
+  , over5
+  , over6
+  , over7
+  , over8
+  , over9
+  , tuple1
+  , tuple10
+  , tuple2
+  , tuple3
+  , tuple4
+  , tuple5
+  , tuple6
+  , tuple7
+  , tuple8
+  , tuple9
+  , uncurry1
+  , uncurry10
+  , uncurry2
+  , uncurry3
+  , uncurry4
+  , uncurry5
+  , uncurry6
+  , uncurry7
+  , uncurry8
+  , uncurry9
+  , (/\)
+  ) as TupleNested
+import Helpers -- Could maybe move this somewhere better:
+  ( (</>)
+  , (<<>>)
+  , (<\>)
+  , appendFirstMaybe
+  , appendLastMaybe
+  , fromJustEff
+  , fromRightEff
+  , liftEither
+  , liftM
+  , liftMWith
+  ) as Helpers
+import Undefined (undefined) as Undefined
+
+-- Imports for extra functions:
+import Prelude
+import Data.Foldable (class Foldable, foldr)
+
+mconcat
+  :: forall (f :: Type -> Type) (m :: Type). Foldable f => Monoid m => f m -> m
+mconcat = foldr (<>) mempty
```

Comment:
`mconcat = Data.Foldable.fold`

Reply:
The output of `Data.Foldable.fold` is very much in Purescript style, although I suspect the Haskellers will a return value as opposed to `Fold { step :: a -> Fold a b, finish :: Unit -> b }`, unless I've misunderstood? @ngua any thoughts on this?

Reply:
Also, this is a left fold on a right associative operator `(<>)`? Why is why I used `foldr`. Hmm.

Reply:
No, I meant like Haskell's `mconcat`. I think we should return the monoidal value, not a `Fold`